src/main/java/com/celements/filebase/uri/FileUriScriptService.java

```diff
@@ -0,0 +1,105 @@
+package com.celements.filebase.uri;
+
+import java.util.Optional;
+
+import javax.annotation.Nullable;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.core.UriBuilder;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.component.annotation.Requirement;
+import org.xwiki.script.service.ScriptService;
+
+import com.celements.filebase.references.FileReference;
+import com.google.common.base.Strings;
+
+import groovy.lang.Singleton;
+
+@Component(FileUriScriptService.NAME)
+@Singleton
+public class FileUriScriptService implements ScriptService {
+
+  public static final String NAME = "fileUri";
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(FileUriScriptService.class);
+
+  @Requirement
+  private FileUriServiceRole fileUriService;
+
+  @NotNull
+  public FileReference createFileReference(@NotEmpty String link) {
+    return FileReference.of(link).build();
+  }
+
+  @NotNull
+  public UriBuilder createFileUrl(@Nullable String fileRef) {
+    return createFileUrl(fileRef, null);
+  }
+
+  @NotNull
+  public UriBuilder createFileUrl(@Nullable String fileRef, @Nullable String action) {
+    return createFileUrl(fileRef, action, null);
+  }
+
+  @NotNull
+  public UriBuilder createFileUrl(@Nullable String fileRef, @Nullable String action,
+      @Nullable String queryString) {
+    if (!Strings.isNullOrEmpty(fileRef)) {
+      return createFileUrl(createFileReference(fileRef), action, queryString);
+    }
+    return UriBuilder.fromPath("");
+  }
+
+  @NotNull
+  public UriBuilder createFileUrl(@Nullable FileReference fileRef, @Nullable String action,
+      @Nullable String queryString) {
+    if (fileRef != null) {
+      try {
+        return fileUriService.createFileUri(fileRef, Optional.ofNullable(action),

```

Comment:
to 1. There is a #try statement for velocity
to 2. Nope URL must be fully qualified. UriBuilder can handle e.g. "/folder/to/file.js" which is not fully quallified and returns a correct toString()

Reply:
1. I was referring to calling `UriBuilder#build` and `URI#toURL` within velocity. the try/catch here won't prevent `MalformedURLException` thrown outside of the try/catch.
2. afaict `UriBuilder#toString` isn't documented/guaranteed behaviour, see [javadoc](https://eclipse-ee4j.github.io/jersey.github.io/apidocs/2.25/jersey/index.html?javax/ws/rs/core/UriBuilder.html).

Reply:
1. I was referring to calling `UriBuilder#build` and `URI#toURL` within velocity. the try/catch here won't prevent `MalformedURLException` thrown outside of the try/catch.
2. it can be misleading calling the SS method names `...Url...` if they don't return Url's. but the service methods are named `...Uri...`. why this inconsistency?